processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -216,18 +217,12 @@ public File mergeQueryableIndex(
       ProgressIndicator progress
   ) throws IOException
   {
+    List<IndexableAdapter> indexAdapteres = new ArrayList<IndexableAdapter>();
+    for (QueryableIndex index : indexes) {
```

Comment:
It's not about the speed of this loop, it's about the speed of access of items in that list. Lists.transform only creates a "view" of the original list, meaning the function gets applied every time you access an element.